File: example/src/test/java/eu/darken/mvpbakery/example/screens/MainPresenterTest.java
package eu.darken.mvpbakery.example.screens;


import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnitRunner;

import static org.mockito.ArgumentMatchers.anyInt;
import static org.mockito.Mockito.never;
import static org.mockito.Mockito.verify;

@RunWith(MockitoJUnitRunner.class)
public class MainPresenterTest {
    @Mock MainPresenter.View view;

    @Test
    public void testCount() throws Throwable {
        MainPresenter mainPresenter = new MainPresenter(null);
        verify(view, never()).showBinderCounter(anyInt());
        mainPresenter.onBindChange(view);
        verify(view).showBinderCounter(1);
        mainPresenter.onBindChange(null);
        mainPresenter.onBindChange(view);
        verify(view).showBinderCounter(2);
    }
}

File: example/src/test/java/eu/darken/mvpbakery/example/screens/MainPresenterTest.kt
package eu.darken.mvpbakery.example.screens


import androidx.fragment.app.Fragment
import io.mockk.MockKAnnotations
import io.mockk.impl.annotations.MockK
import io.mockk.verify
import org.junit.Before
import org.junit.Test


class MainPresenterTest {
    @MockK
    lateinit var view: MainPresenter.View

    @Before
    fun setup() {
        MockKAnnotations.init(this, relaxUnitFun = true)
    }

    @Test
    fun testCount() {
        val mainPresenter = MainPresenter(Fragment::class.java)
        verify(exactly = 0) { view.showBinderCounter(any()) }
        mainPresenter.onBindChange(view)
        verify(exactly = 1) { view.showBinderCounter(any()) }
        mainPresenter.onBindChange(null)
        mainPresenter.onBindChange(view)
        verify(exactly = 2) { view.showBinderCounter(any()) }
    }
}
